reference/reference.go

```diff
@@ -249,21 +249,29 @@ func ParseNamed(s string) (Named, error) {
 	return named, nil
 }
 
-// WithName returns a named object representing the given string. If the input
-// is invalid ErrReferenceInvalidFormat will be returned.
-func WithName(name string) (Named, error) {
```

Comment:
The `WithName()` should probably be kept (but marked "deprecated") to keep the Go API backward compatible? (Given that the repo uses SemVer, and `go mod` would expect any `v1.x` version to be backward compatible)